subtests/docker_cli/systemd/systemd.py

```diff
@@ -65,22 +52,16 @@ def sysd_action(action=None, sysd_unit=None):
             command = '{} {}'.format(command, sysd_unit)
         utils.run(command, ignore_status=False)
 
-    @staticmethod
-    def read_socket(host, port):
-        max_read_tries = 10
-        num_read_try = max_read_tries
-        while num_read_try:
-            # using timeout 13-sec as it's 1-sec longer than two DNS timeouts
-            s = socket.create_connection((host, port), 13)
-            s.settimeout(2.0)
-            try:
-                time_from_socket = s.recv(512).strip()
-                if time_from_socket:
-                    s.shutdown(socket.SHUT_RDWR)
-                    return time_from_socket
-                num_read_try -= 1
-            except socket.timeout:
-                pass
-            time.sleep(1.0)
-        raise DockerTestError("Failed to read from socket with %d tries"
-                              % max_read_tries)
+    def init_unitfile(self, separator=None, changeto=None):
+        tmp_unitfile = '{}.{}'.format(self.sub_stuff['unit_dest'], "tmp")
+        with open(self.sub_stuff['unit_dest'], "r") as srcfile:
```

Comment:
If test dies right here, it will leak a ``/etc/systemd/system/blah.tmp`` file that's never cleaned up.  It would be better to make the file in ``self.tmpdir`` and then ``os.rename()`` it into place.  Anything in ``self.tmpdir`` is automatically cleaned up by autotest.

An alternate approach is to read the entire file into a buffer (a list of lines), modify the buffer, and ``file.writelines()`` it out in place (instead of copying it).  There's a lot less disk I/O involved doing it this way, though you wouldn't really notice with such a small file.

Though for the purposes of this test, either way is fine.